crypto/bio/bio_mem.c

```diff
@@ -81,7 +87,8 @@ BIO *BIO_new_mem_buf(const void *buf, ossl_ssize_t len) {
     return NULL;
   }
 
-  b = (BUF_MEM *)ret->ptr;
+  bbm = (BIO_BUF_MEM *)ret->ptr;
+  b = bbm->buf;
   // BIO_FLAGS_MEM_RDONLY ensures |b->data| is not written to.
   b->data = (void *)buf;
   b->length = size;
```

Comment:
There are several places that we cast `b->length` to an `int`.  We should verify that `size < INT_MAX` here.

Reply:
Looked at the code again. We only cast b->length to an int in `mem_read` and `mem_gets` when the requested number of bytes (which is an `int`) is larger than `b->length`. So I think it's safe not to have this check. 

In `BUF_MEM_append`, the value of `b->length` is also modified, so it's better to have the caster handling the overflow anyways.